File: test-project-mixed/greeter.provider-jdk11/src/main/java/examples/greeter/Friendly.java
package examples.greeter;

import examples.greeter.api.Greeter;
import java.io.*;
import java.util.*;
import javax.annotation.Nonnull;

public class Friendly implements Greeter {
    @Override @Nonnull
    public String hello() {
        var stream = this.getClass().getResourceAsStream("/greeting.txt");
        try (BufferedReader reader = new BufferedReader(new InputStreamReader(stream, "utf-8"))) {
            return reader.readLine();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}

File: test-project-kotlin/greeter.provider/src/main/kotlin/examples/greeter/Friendly.kt
package examples.greeter

import examples.greeter.api.Greeter
import java.io.*
import javax.annotation.concurrent.Immutable;

@Immutable
class Friendly: Greeter {
    override fun hello(): String {
        val stream = javaClass.getResourceAsStream("/greeting.txt")
        val reader = BufferedReader(InputStreamReader(stream, "utf-8"))
        return reader.readLine()
    }
}
